### PR TITLE
Deepest unique naming

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* **BREAKING**: modifies transaction and operation segment/span naming to use deepest *unique* path instead of first deepest path.

  The deepest unique path is the deepest path in the selection set of a query where only one field was selected at each level. 'id' and '__typename' fields are automatically excluded from this decision. For example: `query { libraries { branch booksInStock magazinesInStock }}` will use the path name 'libraries' and  `query { libraries { branch __typename id }}` will use the path name 'libraries.branch'. For more information, see the latest [transaction naming documentation](https://github.com/newrelic/newrelic-node-apollo-server-plugin/blob/main/docs/transactions.md).
  
  This may result in naming changes for your existing GraphQL queries. Any charts, etc. where you are querying by transaction name will need to be updated upon upgrading.

* **BREAKING**: removed the deepest path attribute 'graphql.operation.deepestPath' from operation segments/spans.

## Links

* Closes: https://github.com/newrelic/newrelic-node-apollo-server-plugin/issues/111

## Details

* ~Transaction documentation still needs to be updated.~ 

* Apologies for all the test noise. Needs some further thought but we should consider how we may simplify the tests. Extracting queries + expected parts makes the tests a bit too hard to grok but perhaps we could organize more around queries (instead of output data like transactions/segments) and then have sub tests that assert the expected pieces. Which maybe could mostly share implementation. A thought anyways. While we shouldn't rework transaction naming often, we def have a lot of similar details spread around even with existing shared test files. Made more complicated by the path/operation name used both for segments and transaction name.

### CODE REVIEW

**Review by carlo-808 — COMMENTED** [review]
This all looks good. Looks like the toil was around adjusting the tests. 😄  
I think you need to update the README to remove comments about the deepest-path metric.

**Comment by michaelgoin** [comment]
OK, documentation updates were quick so was able to squeeze those in before taking off. Hopefully caught everything.

File: docs/transactions.md [hunk 1]
```diff
@@ -13,29 +13,96 @@ query {
...
+      issue,
+      title
+    }
+  }
+}
+```
+
+We will select a deepest unique path of 'libraries' as we select multiple fields beyond that point. Any resolver executed beyond that point may contribute to the performance characteristics of the transaction.
+
+If they query were to only select one field per resolver, we select the full path as each selection set is unique.
```
Comment (carlo-808) [inline docs/transactions.md:75]:
```suggestion
If the query were to only select one field per resolver, we select the full path as each selection set is unique.
```

**Review by carlo-808 — APPROVED** [review]
Looks good. I see we will now use deepest unique path for transaction naming. Deepest path metrics have been removed and docs updated. 👍

**Pull request merged**